ONSdigital/ras-party
# Performance improvement 1300 concurrency

# Motivation and Context
Performance Improvement

# What has changed
Changing Gunicorn worker_class from `sync` to `gevent`. The `gevent` workers provide the following benefit.
1.  We no longer see cumulative response times across services. So, if django service is responding in approx. 1s while creating account. The other services are still responding in <1s.
2. Because of point 1 above, create account requests (or any slow request in general) doesn’t affect other concurrent requests. So even when we have hundreds of users creating accounts concurrently, accessing home page (which does not involve the busy django service) takes only up to 50ms on average.


# How to test?
- The changes have been load tested in Prunes. Just check the deployment pipeline is green after changes are merged to master.

# Links
http://docs.gunicorn.org/en/latest/design.html
